helix-core/src/main/java/org/apache/helix/HelixManagerFactory.java

```diff
@@ -61,6 +61,22 @@ public static HelixManager getZKHelixManager(String clusterName, String instance
     return new ZKHelixManager(clusterName, instanceName, type, zkAddr, stateListener);
   }
 
+  /**
+   * Construct a zk-based cluster manager that enforces all types (PARTICIPANT, CONTROLLER, and
+   * SPECTATOR) to have a name
+   * @param clusterName
+   * @param instanceName
+   * @param type
+   * @param zkAddr
+   * @param stateListener
+   * @param helixManagerProperty
+   * @return a HelixManager backed by Zookeeper
+   */
+  public static HelixManager getZKHelixManager(String clusterName, String instanceName,
+      InstanceType type, String zkAddr, HelixManagerStateListener stateListener, HelixManagerProperty helixManagerProperty) {
+    return new ZKHelixManager(clusterName, instanceName, type, zkAddr, stateListener, helixManagerProperty);
```

Comment:
I have concerns over this constructor, and also storing the reset timeout in HelixManagerProperty in general:

If you check the ZkHelixManager logic, you can see that `zkAddr` and `helixManagerProperty.getZkConnectionConfig()` cannot coexist. This is why HelixManagerProperty is only provided in the constructor below this new one: the `zkAddr` in that constructor is null. Therefore, it concerns me that this constructor can lead to a lot of ZkHelixManager initialization failures because of the above mentioned conflict. I believe most users will provide both a zkAddr and a connection config.

Reply:
To add on, `public ZKHelixManager(String clusterName, String instanceName, InstanceType instanceType,
      String zkAddress, HelixManagerStateListener stateListener,
      HelixManagerProperty helixManagerProperty)` is already a public constructor in `ZkHelixManager` for a while.

Reply:
While that constructor has been there for a while, the more common pattern for customers is to create a ZkHelixManager through HelixManagerFactory, and in the factory, the existence of zkAddr and HelixManagerProperty is so far mutually exclusive. 
With this new constructor provided, I predict there will be confusion - it's the first time zkAddr and HelixManagerProperty are both accepted, and we now must make it explicit to the customers that only 1 should be provided. I don't think adding a warning comment will be enough - do you have some ideas about "better logic handling" you mentioned above?
I believe the best way for this problem is not to enforce the RealmAwareConnectionConfig in HelixManagerProperty, but that's out of the scope of this PR. 
With what we have now, I will much prefer the reset value to be set in another function, similar to `setLiveInstanceInfoProvider`, so something like `setMessageHandlerResetTimeout`.

Reply:
The `insert zkAddress into helixManagerProperty` approach is what makes sense to me, but this is again outside the scope. I agree that `setMessageHandlerResetTimeout` will require more work to refactor. 
It's not perfect but I will leave it to your call. The bare minimum I will ask for is a note in the javadoc to highlight that zkaddress should not coexist with the connection config, in case untested code gets deployed to participants and fail.

Reply:
Agree, implemented this one and updated